### PR TITLE
INTYGFV - 14817: Add textfield component in certificate

### PR DESCRIPTION
**Pull request by UlrikaSopra**

### CODE REVIEW

**Comment by SamuelTilly** [comment]
Wooh! first PR @UlrikaSopra nicely done! 🥳🎆🍾

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const input = screen.getByRole('textbox')
+  userEvent.type(input, 'Hello, World!')
+  expect(input).toHaveValue('Hello, World!')
+})
+
+it('disables component if disabled is set', () => {
+  render(<UeTextField question={mockQuestion} disabled={true} />)
+  const input = screen.getByRole('textbox')
+  expect(input).toBeDisabled()
+})
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx:32]: Would like a test that validates that a pre-defined value in `mockQuestion.value.text`, is rendered correctly.

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const useSelectorSpy = jest.spyOn(redux, 'useSelector')
+const useDispatchSpy = jest.spyOn(redux, 'useDispatch')
+
+useSelectorSpy.mockReturnValue({})
+useDispatchSpy.mockReturnValue(jest.fn())
+
+const mockQuestion = {
+  value: { type: CertificateDataValueType.TEXT },
+  config: ({ prop: '' } as unknown) as CertificateDataConfig,
+} as CertificateDataElement
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx:19]: Could use the same `fakeTextFieldElement` function that is used in `UeTextField.stories.tsx`  to generate data.

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  return (
+    <div className="iu-pt-200 iu-grid-cols iu-grid-cols-12">
+      <TextInput
+        className="iu-grid-span-6"
+        disabled={disabled}
+        hasValidationError={questionHasValidationError}
+        onChange={handleChange}
+        name={questionConfig.id}
+        value={text === null ? '' : text}
+        limit={textValidation ? textValidation.limit : 100}></TextInput>
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Inputs/UeTextField.tsx:55]: Could replace `></TextInput>` with `/>`

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
Other than the missing test, I'm just nit-picking on some minor things. Looks good 🥇

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+const useSelectorSpy = jest.spyOn(redux, 'useSelector')
+const useDispatchSpy = jest.spyOn(redux, 'useDispatch')
+
+useSelectorSpy.mockReturnValue({})
+useDispatchSpy.mockReturnValue(jest.fn())
+
+const mockQuestion = {
+  type: fakeTextFieldElement({ id: '1' })['1'],
+}
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx:17]: Please replace with `const mockQuestion = fakeTextFieldElement({ id: '1' })['1']`. `type` is not necessary here.

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx (hunk 1 above)
Comment (UlrikaSopra) [reply packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx:36]: Added a test for default values , hope it was correct :)

File: packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+useSelectorSpy.mockReturnValue({})
+useDispatchSpy.mockReturnValue(jest.fn())
+
+const mockQuestion = fakeTextFieldElement({ id: '1' })['1']
+
+it('renders component with correct default values', () => {
+  render(<UeTextField question={mockQuestion} disabled={false} />)
+  const input = screen.getByRole('textbox')
+  expect(input).toHaveValue('')
+})
```
Comment (SamuelTilly) [inline packages/webcert/src/feature/certificate/Inputs/UeTextField.test.tsx:21]: We need to make sure that the value from `mockQuestion.value.text` is displayed correctly, so we need to add a value to  `mockQuestion` before checking the input value.

Something like this:
```typescript
const mockQuestion = fakeTextFieldElement({ id: '1', { value: { text: 'foo' }} })['1']

it('renders component with correct default values', () => {
  render(<UeTextField question={mockQuestion} disabled={false} />)
  const input = screen.getByRole('textbox')
  expect(input).toHaveValue('foo')
})
```

**Review by SamuelTilly — APPROVED** [review]

**Pull request merged**